wrapper/src/main/java/software/amazon/jdbc/plugin/cache/CacheConnection.java

```diff
@@ -437,16 +529,21 @@ private static RedisURI buildRedisURIStatic(String hostname, int port, boolean u
    */
   static CachePingConnection createPingConnection(String hostname, int port, boolean isReadOnly, boolean useSSL,
       Duration connectionTimeout, boolean iamAuthEnabled, AwsCredentialsProvider credentialsProvider, String cacheIamRegion,
-      String cacheName, String cacheUsername, String cachePassword) {
+      String cacheName, String cacheUsername, String cachePassword, boolean isClusterMode) {
```

Comment:
You don't need to know if you are in cluster mode or not for PING. You can simply use a regular `StatefulRedisConnection` (works for both CMD and CME).